ast/src/parsed/mod.rs

```diff
@@ -140,6 +144,18 @@ impl PilStatement {
                         .map(move |v| (name, Some(&v.name), SymbolCategory::TypeConstructor)),
                 ),
             ),
+            PilStatement::TraitImplementation(
```

Comment:
No, I think the trait implementation should not be accessible at all. It is only accessible through the trait declaration.